### PR TITLE
make sure app state is cleared on user sign off

### PR DESCRIPTION
**Pull request by DmitryEfimenko**
also disable autocomplete on username/password inputs

### CODE REVIEW

File: src/app/auth/components/login.component.html [hunk 1]
```diff
@@ -13,11 +13,11 @@
       <form novalidate [formGroup]="authForm" (ngSubmit)="login()">
         <div>
           <md-input-container>
-            <input md-input formControlName="username" placeholder="User name" required type="text" focused>
+            <input md-input formControlName="username" placeholder="User name" required type="text" autocomplete="off" focused>
```
Comment (abashmak) [inline src/app/auth/components/login.component.html]: In my opinion autocomplete for username should be enabled by default, except for users who are really paranoid. So I think this should be a config setting that is read during build. Can you investigate how to make that happen?

File: src/app/reducers/index.ts [hunk 2]
```diff
@@ -21,6 +21,9 @@ const reducers = {
 };
 
 export function reducer(state: any, action: any) {
+  if (action && action.type === 'USER_LOGOUT') {
+    state = undefined;
+  }
```
Comment (abashmak) [inline src/app/reducers/index.ts]: Just curious, what does this accomplish?

Comment (DmitryEfimenko) [inline src/app/reducers/index.ts]: I chose to use [ngrx/store](https://github.com/ngrx/store) for app state management. This was actually that concept that I wanted to wrap my head around. Basically `state` is the object where all info about user's categories, items, and any other relevant app information would be stored. Setting it to undefined assures that other user, after logging in will not get access to previous user's info.

File: src/app/auth/components/login.component.html (hunk 1 above)
Comment (DmitryEfimenko) [inline src/app/auth/components/login.component.html]: made it configurable

**Pull request merged**